### PR TITLE
Update l10n branching strategy

### PR DESCRIPTION
**Pull request by gochist**
To explain how sig-docs-ko-owners team works in real for #13850 

/cc @zacharysarah

### CODE REVIEW

**Comment by zacharysarah** [comment]
@gochist Thanks for this PR! 👍 Do you want to wait to merge this until #13850 merges?

**Comment by gochist** [comment]
@zacharysarah It is okay to move the changes in this PR to #13850. I just opened this PR to share my idea.

File: content/en/docs/contribute/localization.md [hunk 1]
```diff
@@ -128,7 +128,7 @@ Localizations must use English files from the most recent release as their sourc
 To find source files for the most recent release:
 
 1. Navigate to the Kubernetes website repository at https://github.com/kubernetes/website.
-2. Select the `release-1.X` branch for the most recent version.
+2. Select the most recent localization development branch or open it. Usually a team member of [@kubernetes/sig-docs-l10n-admins] opens the development branch from `master` branch on https://github.com/kubernetes/website. The `master` branch will be merged onto the most recent release branch eventually. See the [branching strategy](#branching-strategy) for more information.
```
Comment (zacharysarah) [inline content/en/docs/contribute/localization.md]: @gochist This section deals with finding source files in English, so it doesn't make sense to add instructions here about opening a development branch. I think your changes on line 172 👇 make a better place for this sort of expansion.

File: content/en/docs/contribute/localization.md [hunk 2]
```diff
@@ -183,15 +183,17 @@ To collaborate on a development branch:
...
 
 Repeat steps 1-4 as needed until the localization is complete. For example, subsequent German development branches would be: `dev-1.12-de.2`, `dev-1.12-de.3`, etc.
 
 Teams must merge localized content into the same release branch from which the content was sourced. For example, a development branch sourced from {{< release-branch >}} must be based on {{< release-branch >}}.
 
 An approver must maintain a development branch by keeping it current with its source branch and resolving merge conflicts. The longer a development branch stays open, the more maintenance it typically requires. Consider periodically merging development branches and opening new ones, rather than maintaining one extremely long-running development branch.
 
-While only approvers can merge pull requests, anyone can open a pull request for a new development branch. No special permissions are required.
+On the beginning of every team milestone, a team member usually opens an issue to update outdated contents by comparing upstream changes between the previous development branch and the current development branch.
```
Comment (zacharysarah) [inline content/en/docs/contribute/localization.md]: @gochist Do you have a gist of the script you use for comparison? Would you please provide a link?

Comment (gochist) [reply content/en/docs/contribute/localization.md]: @zacharysarah Here is the link, https://gist.github.com/gochist/8654460717b4bbb4ac490dda79a547fa

**Pull request closed without merging**